### PR TITLE
Log in using social networks (G+) and Apply Spring Security to authorization process

### PR DESCRIPTION
**Pull request by LostFan**

### CODE REVIEW

File: src/main/java/com/epam/idea/core/model/SocialMediaService.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.epam.idea.core.model;
+
+/**
+ * Created by Ihar_Niakhlebau on 08-Jun-15.
+ */
+public enum SocialMediaService {
+	NONE(""), GOOGLE("GOOGLE"), FACEBOOK("FACEBOOK"), VK("VK");
```
Comment (Hanna-Aliakseichykava) [inline src/main/java/com/epam/idea/core/model/SocialMediaService.java:7]: +

File: src/main/java/com/epam/idea/core/social/RegistrationForm.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+import com.epam.idea.core.model.SocialMediaService;
+import com.epam.idea.core.social.annotation.PasswordsNotEmpty;
+import com.epam.idea.core.social.annotation.PasswordsNotEqual;
+import org.hibernate.validator.constraints.Email;
+import org.hibernate.validator.constraints.NotEmpty;
+
+import javax.validation.constraints.Size;
+
+@PasswordsNotEmpty(
```
Comment (Hanna-Aliakseichykava) [inline src/main/java/com/epam/idea/core/social/RegistrationForm.java]: +

File: src/main/java/com/epam/idea/core/social/util/SecurityUtil.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import com.epam.idea.core.model.CommonUserDetails;
+import com.epam.idea.core.model.User;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+public class SecurityUtil {
+
+	public static void logInUser(User user) {
+		CommonUserDetails userDetails = CommonUserDetails.getBuilder()
```
Comment (Hanna-Aliakseichykava) [inline src/main/java/com/epam/idea/core/social/util/SecurityUtil.java:11]: May be, it makes sense to have method for simple case, that takes only user and creates CommonUserDetails

File: src/main/java/com/epam/idea/core/social/util/ValidatorUtil.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package com.epam.idea.core.social.util;
+
+import javax.validation.ConstraintValidatorContext;
+import java.lang.reflect.Field;
+
+public class ValidatorUtil {
+
```
Comment (Hanna-Aliakseichykava) [inline src/main/java/com/epam/idea/core/social/util/ValidatorUtil.java:7]: +

File: src/main/java/com/epam/idea/rest/config/SocialConfig.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+import org.springframework.social.security.AuthenticationNameUserIdSource;
+
+/**
+ * Created by Ihar_Niakhlebau on 04-Jun-15.
+ */
+@Configuration
+@EnableSocial
+public class SocialConfig implements SocialConfigurer {
+
+	public static String clientId = "133307390078-irhbk08u5omsjfra8qorte4qgkqsvuuv.apps.googleusercontent.com";
```
Comment (Hanna-Aliakseichykava) [inline src/main/java/com/epam/idea/rest/config/SocialConfig.java]: It may be worth adding word 'google' to variables' names (clientId, clientSecret)

Comment (LostFan) [inline src/main/java/com/epam/idea/rest/config/SocialConfig.java]: Ok, I will rename in the next commit.

File: src/test/java/com/epam/idea/rest/controller/UserControllerTest.java [hunk 6]
```diff
@@ -528,123 +518,123 @@ public void shouldCreateIdeaWithAuthorUserAndReturnItWithHttpCode201() throws Ex
...
+//				.accept(APPLICATION_JSON_UTF8))
+//				.andExpect(status().isNotFound())
+//				.andExpect(content().contentType(APPLICATION_JSON_UTF8))
+//				.andExpect(jsonPath("$", hasSize(1)))
+//				.andExpect(jsonPath("$[0].logref").value(USER_NOT_FOUND_LOGREF))
+//				.andExpect(jsonPath("$[0].message").value("Could not find session with id: 1."));
+//
+//		verify(this.userSessionServiceMock, times(1)).findOne(DEFAULT_USER_SESSION_ID);
+//		verifyNoMoreInteractions(this.userSessionServiceMock);
+//	}
```
Comment (Hanna-Aliakseichykava) [inline src/test/java/com/epam/idea/rest/controller/UserControllerTest.java:638]: Please, add more tests (unit and ui)

**Comment by Hanna-Aliakseichykava** [comment]
:+1:

**Comment by alxx88** [comment]
:+1:

**Pull request merged**